doc/source/conf.py

```diff
@@ -408,14 +397,18 @@ def linkcode_resolve(domain, info):
 
     if fn.startswith('scipy/'):
         m = re.match(r'^.*dev0\+([a-f0-9]+)$', scipy.__version__)
+        base_url = "https://github.com/scipy/scipy/blob"
         if m:
-            return "https://github.com/scipy/scipy/blob/%s/%s%s" % (
-                m.group(1), fn, linespec)
+            return f"{base_url}/{m.group(1)}/{fn}{linespec}"
         elif 'dev' in scipy.__version__:
-            return "https://github.com/scipy/scipy/blob/main/%s%s" % (
-                fn, linespec)
+            return f"{base_url}/main/{fn}{linespec}"
         else:
-            return "https://github.com/scipy/scipy/blob/v%s/%s%s" % (
-                scipy.__version__, fn, linespec)
+            return f"{base_url}/v{scipy.__version__}/{fn}{linespec}"
     else:
         return None
+
+
+# Tell overwrite numpydoc's logic to render examples containing rng.
+SphinxDocString._str_examples = _rng_html_rewrite(
+    SphinxDocString._str_examples
```

Comment:
Where does this get called?

Reply:
This is the magic part! When numpydoc is rendering the docstrings (so called by Sphinx), various methods of `SphinxDocString` are used depending on what part of the docstrings is being processed. Here we overwrite `_str_examples` (which only process the example section of a docstrings) to have the RNG logic.